filebeat/prospector/redis/harvester.go

```diff
@@ -0,0 +1,131 @@
+package redis
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/elastic/beats/filebeat/harvester"
+	rd "github.com/garyburd/redigo/redis"
+	"github.com/satori/go.uuid"
+)
+
+type Harvester struct {
+	outlet    channel.Outleter
+	id        uuid.UUID
+	done      chan struct{}
+	conn      rd.Conn
+	forwarder harvester.Forwarder
+}
+
+// log contains all data related to one slowlog entry
+//
+// 	The data is in the following format:
+// 	1) (integer) 13
+// 	2) (integer) 1309448128
+// 	3) (integer) 30
+// 	4) 1) "slowlog"
+// 	   2) "get"
+// 	   3) "100"
+//
+type log struct {
+	id        int64
+	timestamp int64
+	duration  int
+	cmd       string
+	key       string
+	args      []string
+}
+
+func NewHarvester(outlet channel.Outleter, conn rd.Conn) *Harvester {
+	return &Harvester{
+		id:     uuid.NewV4(),
+		done:   make(chan struct{}),
+		outlet: outlet, // All harvester can use the same outlet as not state
+		conn:   conn,
+	}
+}
+
+func (h *Harvester) Start() error {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Harvester.Start should have comment or be unexported